artpaint/paintwindow/PaintWindow.cpp

```diff
@@ -316,15 +291,27 @@ PaintWindow::PaintWindow(BRect frame, const char* name, uint32 views,
 		PostMessage(&msg, this);
 	}
 
+	BGroupLayout* layout = BLayoutBuilder::Group<>(this, B_VERTICAL, 0)
+//		.Add(fMenubar)
+		.AddGroup(B_HORIZONTAL, 0)
+			.AddGroup(B_VERTICAL, 0)
+				.Add(fBackground)
+//				.Add(fStatusView, 0)
+				.End()
+			.Add(fVerticalScrollbar)
+			.End()
+		.Add(fHorizontalScrollbar);
+
+	if (fMenubar) layout->AddView(0, fMenubar);
+
+	if (fStatusView)
+		((BGroupLayout*) ((BGroupLayout*) layout->ItemAt(fMenubar != NULL))->ItemAt(0))->AddView(fStatusView, 0);
```

Comment:
Thanks! I should have thought of the first bullet.

I prefer your 2nd bullet to my way and tried the follwing, which replaces lines 294-308:
`
BGroupLayout* inner = new BGroupLayout(B_VERTICAL, 0);
inner->AddView(fBackground);
if (fStatusView) inner->AddView(fStatusView, 0);

BGroupLayout* outer = BLayoutBuilder::Group<>(this, B_VERTICAL, 0)
//	.Add(fMenubar)
	.AddGroup(B_HORIZONTAL, 0)
		.Add(inner)
		.Add(fVerticalScrollbar)
		.End()
	.Add(fHorizontalScrollbar);

if (fMenubar) outer->AddView(0, fMenubar);
`
but I got this:
![add](https://cloud.githubusercontent.com/assets/6434360/23198026/522e868a-f879-11e6-8ab4-0a2446be7bf9.png)

When I changed `.Add(inner)` to `.AddGroup(inner)`, I got this:
![addgroup](https://cloud.githubusercontent.com/assets/6434360/23198079/cf8f35d4-f879-11e6-9f97-3018baf581e2.png)